### DISCUSSION TITLE
Using Perses Backend as API for Dashboard and Datasource Files

### DISCUSSION THREAD

**Discussion by rasmusvan** [perses/perses#2131] [Q&A] [answered]
This is a follow-up to [issue #2108](https://github.com/perses/perses/issues/2108). Based on the information provided there, I have created a minimal reproduction repository for embedding a dashboard, available [here](https://github.com/rasmusvan/perses-embedded-dashboard-mrr) on the “embed-dashboard-mrr” branch.

The current issue I’m facing is retrieving the dashboard and datasource manifest files from the Perses backend (usually accessed via percli) instead of storing them locally. Essentially, I want to use the Perses backend as an API for these datasource/dashboard files. In the repository, I’m currently mocking them with fakeDashboard and fakeDatasource. Is there a way to achieve this using the Perses backend?

**Comment by Nexucis** [comment, marked as answer]
Hello,

The Perses backend is providing a REST API that allows you to retrieve dashboard, project, datasource, whatever Perses is supporting.
API documentation is available on our website. For example, for the dashboard: https://perses.dev/docs/perses/v0.46.0/api/dashboard.md/

**Reply by rasmusvan** [reply]
I managed to get querying datasources to work, because they are only used in DatasourceApiImpl, which expects promises like Promise\<GlobalDatasource\>. However when I try to query a dashboard, it's not as trivial because the DashboardStoreProps requires a DashboardResource, not a Promise\<DashboardResource\>. Because of that I am getting errors like "Conversion of type 'Promise\<DashboardResource\>' to type 'DashboardResource' may be a mistake because neither type sufficiently overlaps with the other." What would be a sensible fix for this?

**Reply by Nexucis** [reply]
I am sorry, I have no idea what you are talking about.

You wanted to know if it was possible to retrieve the dashboard from the Perses backend. And I have told you, yes it was possible by consuming the REST API exposed, using the client you like (CLI, CURL, programmatic request call).

For example: 

```bash
curl -XGET `http://localhost:8080/api/v1/dashboards
```

 will return a list of dashboard regardless the project where they are stored.

So I think I have answered your question. For the rest about what you are talking, it's on your side and depends only on what you are developing.
